### PR TITLE
cron CI: various fixes

### PR DESCRIPTION
**Pull request by stefantalpalaru**
I tried to make it compile again with Nim-1.4 but I couldn't find a workaround for this regression: https://github.com/nim-lang/Nim/issues/17742

It hits us in https://github.com/status-im/nimbus-eth2/blob/8b15c5d75f55007635cb1656b5c525196336847c/tests/test_ssz_roundtrip.nim

Nim devel, on the other hand, fails with this bizzarre error:

```text
/mnt/sde1/storage/nimbus-eth2/beacon_chain/eth1/eth1_monitor.nim(437, 59) template/generic instantiation of `init` from here
/mnt/sde1/storage/nimbus-eth2/vendor/nimbus-build-system/vendor/Nim/lib/pure/collections/sequtils.nim(966, 20) Error: cannot attach a custom pragma to 's2'
```

### CODE REVIEW

File: beacon_chain/networking/eth2_network.nim [hunk 1]
```diff
@@ -1271,7 +1271,7 @@ proc p2pProtocolBackendImpl*(p: P2PProtocol): Backend =
   result.SerializationFormat = Format
   result.RequestResultsWrapper = ident "NetRes"
 
-  result.implementMsg = proc (msg: p2p_protocol_dsl.Message) =
+  result.implementMsg = proc (msg: p2p_protocol_dsl.Message) {.raises: [Exception].} =
```
Comment (arnetheduck) [inline beacon_chain/networking/eth2_network.nim]: this looks wrong and needs fixing in `nim-eth` first - ie `implementMsg` needs to make up its mind

File: config.nims [hunk 2]
```diff
@@ -73,6 +73,11 @@ switch("passL", "-fno-omit-frame-pointer")
 # for heap-usage-by-instance-type metrics and object base-type strings
 --define:nimTypeNames
 
+when (NimMajor, NimMinor, NimPatch) >= (1, 4, 0):
+  # "Error: 't' is of type <lent string> which cannot be captured as it would
+  # violate memory safety"
+  --define:nimWorkaround14447
```
Comment (arnetheduck) [inline config.nims]: let's avoid these

Comment (stefantalpalaru) [reply config.nims]: How?

Comment (arnetheduck) [reply config.nims]: by fixing the underlying root cause or not supporting 1.4

Comment (arnetheduck) [reply config.nims]: these workarounds end up rotting in upstream - we don't ever want to be stuck in a situation where we depend on them - instead, it's ok that the 1.4 compiles show a failure - this is the honest state for that version of the compiler and highlights what issues need to be fixed in the future

Comment (stefantalpalaru) [reply config.nims]: Workaround removed and corresponding libp2p issue created.

**Review by arnetheduck — APPROVED** [review]

**Pull request merged**